frontend/src/main/scala/bloop/Cli.scala

```diff
@@ -139,29 +144,22 @@ object Cli {
         .getOrElse(cliOptions.common.workingPath.resolve(".bloop-config"))
     }
 
-    import bloop.engine.{State, Build}
-    def loadStateFor(configDirectory: AbsolutePath, logger: Logger): State = {
-      State.stateCache.getStateFor(configDirectory) match {
-        case Some(state) => state
-        case None =>
-          State.stateCache.addIfMissing(configDirectory, path => {
-            val projects = Project.fromDir(configDirectory, logger)
-            val build: Build = Build(configDirectory, projects)
-            State(build, logger)
-          })
-      }
+    def createLogger(configDir: AbsolutePath, options: CliOptions, logToFile: Boolean): Logger = {
+      val loggerId = configDir.syntax
+      if (logToFile) BloopLogger.atFile(loggerId, Paths.bloopLogsDir.resolve("bsp.log"))
+      else BloopLogger.at(loggerId, options.common.out, options.common.err)
     }
 
-    val cliOptions = action match {
-      case e: Exit => CliOptions.default
-      case p: Print => CliOptions.default
-      case r: Run => r.command.cliOptions
+    val (cliOptions, logToFile) = action match {
+      case e: Exit => (CliOptions.default, false)
+      case p: Print => (CliOptions.default, false)
+      case Run(cmd: Commands.Bsp, _) => (cmd.cliOptions, false)
+      case r: Run => (r.command.cliOptions, false)
```

Comment:
`logToFile` is always false?

Reply:
Yup, because of the above reason :smile:.